app/controllers/mixins/actions/vm_actions/reconfigure.rb

```diff
@@ -338,16 +350,12 @@ def build_vmcdrom_list(vm)
           end
         end
 
-        def supports_reconfigure_disksize?
-          @reconfigitems && @reconfigitems.size == 1 && @reconfigitems.first.supports_reconfigure_disksize? && @reconfigitems.first.supports_reconfigure_disks?
-        end
-
-        def supports_reconfigure_network_adapters?
-          @reconfigitems && @reconfigitems.size == 1 && @reconfigitems.first.supports_reconfigure_network_adapters?
-        end
-
-        def supports_reconfigure_cdroms?
-          @reconfigitems && @reconfigitems.size == 1 && @reconfigitems.first.supports_reconfigure_cdroms?
+        # ui helper method
+        def item_supports?(*features)
+          # this one is a special case
+          features << :reconfigure_disks if features.include?(:reconfigure_disksize)
+          @reconfigitems && @reconfigitems.size == 1 &&
+            features.all? { |feature| @reconfigitems.first.supports?(feature) }
```

Comment:
Possible refactor?

```suggestion
          item = @reconfigitems.first if @reconfigitems&.size == 1
          item && features.all? { |feature| item.supports?(feature) }
```

Reply:
Aside, if it's common to check for multiple features, I wonder if it would be helpful for `supports?` to accept an Array?  Then this code would simplify to something like

```ruby
item = @reconfigitems.first if @reconfigitems&.size == 1
item&.supports?(features)
```